bsmetadata/preprocessing_tools/wikipedia_desc_utils.py

```diff
@@ -40,3 +43,17 @@ def fetch_website_description_from_keyword(self, keyword: str) -> Optional:
             self.cache[keyword] = self.extract_wiki_desc(keyword)
 
         return self.cache[keyword]
+
+    def fetch_entity_description_from_keyword(self, keyword: str) -> str:
+        try:
+            key = string.capwords(keyword)
+            text = self.wiki_dump_db.get_paragraphs(key)[0].text
+            text = re.sub(r"\((?:[^)(]|\([^)(]*\))*\)", "", text)
+            text = nltk.sent_tokenize(text)[0]
+        except KeyError:
+            try:
+                text = self.wiki_dump_db.get_paragraphs(self.redirects_map[keyword])[0].text
+                text = nltk.tokenize.sent_tokenize(text)[0]
+            except KeyError:
+                text = ""
+        return text
```

Comment:
Can you detail a bit more which line is likely to return which error? 

I wonder if there is a way to de-interlace the exceptions. Or even to use the `fetch_wikipedia_description_for_title` method. 

Also, why in the second try/except, the `re.sub(r"\((?:[^)(]|\([^)(]*\))*\)", "", text)` processing is not necessary anymore?

If ever it is so interesting to have the second try/except I think the following formulation could bring clarity:

```suggestion
    def fetch_entity_description_from_keyword(self, keyword: str) -> str:
        key = string.capwords(keyword)
        text = self.fetch_wikipedia_description_for_title(key)

        if text is not None:
            key = self.redirects_map[keyword]
            text = self.fetch_wikipedia_description_for_title(key)
        
        if text is None:
            text = ""
        
        return text
```

Reply:
@shanyas10 , I'd love to hear your opinion on this too. Wouldn't it make sense to call the "key" variables here "title"?

Reply:
Hi @SaulLu , yes indeed it would make more sense to call the "key" variables as title. Also having text = re.sub(r"\((?:[^)(]|\([^)(]*\))*\)", "", text) before returning the text would ensure that we don't have unnecessary tokens pulled from wikipedia descriptions.

Reply:
Thank you so much for sharing this info! 

@manandey , do you agree with renaming these variables?

Reply:
@SaulLu, I have edited my previous comment. :)

Reply:
@SaulLu, I have updated `key` to `title` as per your advice. :) Regarding using `fetch_wikipedia_description_for_title` method, I was not preferring to use it because in case it is unable to find any text for that `title/keyword`, it will return `None` and I prefer to have `""` instead in such a case.

Reply:
> fetch_wikipedia_description_for_title method, I was not preferring to use it because in case it is unable to find any text for that title/keyword, it will return None and I prefer to have "" instead in such a case.

Yes absolutely, I was just proposing to use the method inside yours :slightly_smiling_face: